soc/arm/nordic_nrf/nrf53/soc.c

```diff
@@ -137,12 +143,177 @@ bool z_arm_on_enter_cpu_idle(void)
 		suppress_message = true;
 	}
 #endif
+#if defined(CONFIG_SOC_NRF53_RTC_PRETICK) && defined(CONFIG_SOC_NRF5340_CPUNET)
+	if (ok_to_sleep) {
+		extern uint32_t rtc_pretick_rtc1_ticks_to_next_event_get(uint32_t cntr);
+
+		uint32_t rtc1_counter = nrf_rtc_counter_get(NRF_RTC1);
+		uint32_t rtc1_ticks_to_next_event = rtc_pretick_rtc1_ticks_to_next_event_get(rtc1_counter);
+		uint32_t rtc1_pretick_cc_val = (rtc1_counter + rtc1_ticks_to_next_event - 1U) & NRF_RTC_COUNTER_MAX;
+		static bool rtc1_pretick_cc_set_on_time = false;
+
+		if (rtc1_pretick_cc_val != nrf_rtc_cc_get(NRF_RTC1, RTC1_PRETICK_CC_CHAN)) {
+			/* The CC for pretick needs to be updated. */
+			nrf_rtc_cc_set(NRF_RTC1, RTC1_PRETICK_CC_CHAN, rtc1_pretick_cc_val);
+
+			/* Let's check if we updated CC on time so that the CC can take effect. */
+			rtc1_counter = nrf_rtc_counter_get(NRF_RTC1);
+			if (((rtc1_pretick_cc_val - rtc1_counter) & NRF_RTC_COUNTER_MAX) < 2) {
```

Comment:
based on this @anangl comment: https://github.com/zephyrproject-rtos/zephyr/blob/0933a4b2949889ca924b10aff05b1093dabe7ba0/drivers/timer/nrf_rtc_timer.c#L242
it probably should be `< 3`. @anangl can you confirm?

Reply:
fixed to `3`